File: android/revenuedog/src/main/kotlin/com/revenuedog/purchases/FetchPolicy.kt
package com.revenuedog.purchases

/** Controls how [Purchases.getCustomerInfo] uses the local cache. */
enum class FetchPolicy {
    /** Return the cached value when it is less than 5 minutes old, otherwise hit the network. */
    CACHED_OR_FETCH,

    /** Always fetch from the network. */
    FETCH_CURRENT,

    /** Only return the cached value; throws [PurchasesError] when no cache exists. */
    CACHE_ONLY
}

File: ios/Sources/RevenueDog/FetchPolicy.swift
import Foundation

/// Controls how ``Purchases/getCustomerInfo(fetchPolicy:)`` resolves
/// ``CustomerInfo`` between the local cache and the network.
public enum FetchPolicy: Sendable, Equatable {
    /// Return the cached value if it is younger than 5 minutes, otherwise hit
    /// the network. This is the default.
    case cachedOrFetch
    /// Always fetch a fresh value from the backend.
    case fetchCurrent
    /// Return the cached value if present and never hit the network. If no
    /// cache exists the SDK falls back to a network fetch (so a value is
    /// always returned).
    case cacheOnly

    /// The default policy used by ``Purchases/getCustomerInfo()``.
    public static let `default`: FetchPolicy = .cachedOrFetch
}
